apps/users-api/src/routes/loginUser.ts

```diff
@@ -0,0 +1,42 @@
+import { Router } from 'express';
+import { z } from 'zod';
+import bcrypt from 'bcrypt';
+import { prisma } from '@competition-manager/prisma';
+import { parseRequest, generateAccessToken, generateRefreshToken } from '@competition-manager/utils';
+
+export const router = Router();
+
+const Body$ = z.object({
+    email: z.string({ message: 'Email must be a string' }).email({ message: 'Email must be a valid email' }),
+    password: z.string({ message: 'Password must be a string' })
+});
+
+router.post(
+    '/login',
+    parseRequest('body', Body$),
+    async (req, res) => {
+        const { email, password } = Body$.parse(req.body);
+        const user = await prisma.user.findUnique({
+            where: {
+                email: email
+            }
+        });
+        if (!user) {
+            res.status(401).json({ message: 'Invalid credentials' });
+            return;
+        }
+        const valid = await bcrypt.compare(password, user.password);
+        if (!valid) {
+            res.status(401).json({ message: 'Invalid credentials' });
+            return;
+        }
+        const accessToken = generateAccessToken(email);
+        const refreshToken = generateRefreshToken(email);
```

Comment:
Ok pour la data à mettre dans le token. Je pense qu'il faudrait créer un zod schema (pas dans shared-package je pense) dans un file TokenData.ts (pas de le folder routes)
Pour pouvoir facilement rajouter des champs dans le token (exemple le role du user)

Reply:
et une fonction `getTokenDataFromUser(user)`

Reply:
TokenData.ts dans shared-package in schemas

Reply:
Pcq besoin dans le front